### ISSUE TITLE
[GLPI 10] Cannot declare class Glpi\Socket

### ISSUE THREAD

**Issue by ukkt** [glpi-project/glpi#10130]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

10.0.0_dev

### Bug description

Hello,

I get the error message "Fatal error: Cannot declare class Glpi\Socket, because the name is already in use in C:\xampp\htdocs\glpi10\src\Socket.php on line 0" when I go to computers and I get no list. For monitors it is ok.

### Relevant log output

```shell
[2021-12-15 18:07:20] glpiphplog.ALERT:   *** PHP Compile Error (64): Cannot declare class Glpi\Socket, because the name is already in use in C:\xampp\htdocs\glpi10\src\Socket.php at line 0  
[2021-12-15 18:07:27] glpiphplog.ALERT:   *** PHP Compile Error (64): Cannot declare class Glpi\Socket, because the name is already in use in C:\xampp\htdocs\glpi10\src\Socket.php at line 0  
[2021-12-15 18:07:29] glpiphplog.ALERT:   *** PHP Compile Error (64): Cannot declare class Glpi\Socket, because the name is already in use in C:\xampp\htdocs\glpi10\src\Socket.php at line 0  
[2021-12-15 18:07:32] glpiphplog.ALERT:   *** PHP Compile Error (64): Cannot declare class Glpi\Socket, because the name is already in use in C:\xampp\htdocs\glpi10\src\Socket.php at line 0  
[2021-12-15 18:07:35] glpiphplog.ALERT:   *** PHP Compile Error (64): Cannot declare class Glpi\Socket, because the name is already in use in C:\xampp\htdocs\glpi10\src\Socket.php at line 0  
[2021-12-15 18:07:37] glpiphplog.ALERT:   *** PHP Compile Error (64): Cannot declare class Glpi\Socket, because the name is already in use in C:\xampp\htdocs\glpi10\src\Socket.php at line 0  
[2021-12-15 18:07:39] glpiphplog.ALERT:   *** PHP Compile Error (64): Cannot declare class Glpi\Socket, because the name is already in use in C:\xampp\htdocs\glpi10\src\Socket.php at line 0  
[2021-12-15 18:07:49] glpiphplog.ALERT:   *** PHP Compile Error (64): Cannot declare class Glpi\Socket, because the name is already in use in C:\xampp\htdocs\glpi10\src\Socket.php at line 0  
[2021-12-15 18:08:09] glpiphplog.ALERT:   *** PHP Compile Error (64): Cannot declare class Glpi\Socket, because the name is already in use in C:\xampp\htdocs\glpi10\src\Socket.php at line 0  
[2021-12-15 18:08:14] glpiphplog.WARNING: Test logger  
[2021-12-15 18:08:32] glpiphplog.ALERT:   *** PHP Compile Error (64): Cannot declare class Glpi\Socket, because the name is already in use in C:\xampp\htdocs\glpi10\src\Socket.php at line 0  
[2021-12-15 18:08:42] glpiphplog.ALERT:   *** PHP Compile Error (64): Cannot declare class Glpi\Socket, because the name is already in use in C:\xampp\htdocs\glpi10\src\Socket.php at line 0  
[2021-12-15 18:08:48] glpiphplog.ALERT:   *** PHP Compile Error (64): Cannot declare class Glpi\Socket, because the name is already in use in C:\xampp\htdocs\glpi10\src\Socket.php at line 0  
[2021-12-15 18:09:49] glpiphplog.ALERT:   *** PHP Compile Error (64): Cannot declare class Glpi\Socket, because the name is already in use in C:\xampp\htdocs\glpi10\src\Socket.php at line 0  
[2021-12-15 18:09:53] glpiphplog.ALERT:   *** PHP Compile Error (64): Cannot declare class Glpi\Socket, because the name is already in use in C:\xampp\htdocs\glpi10\src\Socket.php at line 0  
[2021-12-15 18:10:41] glpiphplog.ALERT:   *** PHP Compile Error (64): Cannot declare class Glpi\Socket, because the name is already in use in C:\xampp\htdocs\glpi10\src\Socket.php at line 0  
[2021-12-15 18:13:14] glpiphplog.WARNING: Test logger  
[2021-12-15 18:14:05] glpiphplog.ALERT:   *** PHP Compile Error (64): Cannot declare class Glpi\Socket, because the name is already in use in C:\xampp\htdocs\glpi10\src\Socket.php at line 0
```

### Page URL

http://localhost/glpi/front/computer.php

### Steps To reproduce

Goto Assets --> Computer

### Your GLPI setup information

<html>
<body>

GLPI 10.0.0-dev (/glpi10 => C:\xampp\htdocs\glpi10) Installation mode: TARBALL Current language:de_DE
--

Operating system: Windows NT KABEL254 10.0 build 14393 (Windows Server 2016) i586 PHP 7.4.26 apache2handler (Core, PDO, Phar, Reflection, SPL, SimpleXML, Zend OPcache, apache2handler, apcu, bcmath, bz2, 	calendar, ctype, curl, date, dom, exif, fileinfo, filter, ftp, gd, gettext, hash, iconv, imap, intl, json, ldap, libxml, 	mbstring, mysqli, mysqlnd, openssl, pcre, pdo_mysql, pdo_sqlite, readline, session, snmp, sodium, standard, tokenizer, xml, 	xmlreader, xmlrpc, xmlwriter, zip, zlib) Setup: max_execution_time="30" memory_limit="128M" post_max_size="8M" safe_mode="" session.save_handler="files" 	upload_max_filesize="2M"  Software: Apache/2.4.33 (Win32) OpenSSL/1.1.1l PHP/7.4.26 (Apache/2.4.33 (Win32) OpenSSL/1.1.1l PHP/7.4.26 Server at kabel254 	Port 80 ) 	Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/96.0.4664.110 Safari/537.36 Edg/96.0.1054.57 Server Software: MySQL Enterprise Server - Commercial 	Server Version: 8.0.11-commercial 	Server SQL Mode: STRICT_TRANS_TABLES,NO_ENGINE_SUBSTITUTION 	Parameters: root@kabel254/glpi10 	Host info: kabel254 via TCP/IP 	 PHP version (7.4.26) is supported. Sessions configuration is OK. Allocated memory is sufficient. mysqli extension is installed. Following extensions are installed: dom, fileinfo, json, simplexml. curl extension is installed. gd extension is installed. intl extension is installed. libxml extension is installed. zlib extension is installed. Database engine version (8.0.11) is supported. The log file has been created successfully. Write access to C:\xampp\htdocs\glpi10/files/_cache has been validated. Write access to C:\xampp\htdocs\glpi10/config has been validated. Write access to C:\xampp\htdocs\glpi10/files/_cron has been validated. Write access to C:\xampp\htdocs\glpi10/files has been validated. Write access to C:\xampp\htdocs\glpi10/files/_dumps has been validated. Write access to C:\xampp\htdocs\glpi10/files/_graphs has been validated. Write access to C:\xampp\htdocs\glpi10/files/_lock has been validated. Write access to C:\xampp\htdocs\glpi10/files/_pictures has been validated. Write access to C:\xampp\htdocs\glpi10/files/_plugins has been validated. Write access to C:\xampp\htdocs\glpi10/files/_rss has been validated. Write access to C:\xampp\htdocs\glpi10/files/_sessions has been validated. Write access to C:\xampp\htdocs\glpi10/files/_tmp has been validated. Write access to C:\xampp\htdocs\glpi10/files/_uploads has been validated. Web access to files directory is protected exif extension is installed. ldap extension is installed. openssl extension is installed. zip extension is installed. bz2 extension is installed. Zend OPcache extension is installed. Following extensions are installed: ctype, iconv, mbstring, sodium. Write access to C:\xampp\htdocs\glpi10/marketplace has been validated. Timezones seems not loaded, see https://glpi-install.readthedocs.io/en/latest/timezones.html.

GLPI_ROOT: C:\xampp\htdocs\glpi10 GLPI_CONFIG_DIR: C:\xampp\htdocs\glpi10/config GLPI_VAR_DIR: C:\xampp\htdocs\glpi10/files GLPI_MARKETPLACE_DIR: C:\xampp\htdocs\glpi10/marketplace GLPI_USE_CSRF_CHECK: 1 GLPI_CSRF_EXPIRES: 7200 GLPI_CSRF_MAX_TOKENS: 100 GLPI_USE_IDOR_CHECK: 1 GLPI_IDOR_EXPIRES: 7200 GLPI_ALLOW_IFRAME_IN_RICH_TEXT:  GLPI_TELEMETRY_URI: https://telemetry.glpi-project.org GLPI_INSTALL_MODE: TARBALL GLPI_NETWORK_MAIL: glpi@teclib.com GLPI_NETWORK_SERVICES: https://services.glpi-network.com GLPI_MARKETPLACE_ALLOW_OVERRIDE: 1 GLPI_MARKETPLACE_MANUAL_DOWNLOADS: 1 GLPI_USER_AGENT_EXTRA_COMMENTS:  GLPI_DISABLE_ONLY_FULL_GROUP_BY_SQL_MODE: 1 GLPI_AJAX_DASHBOARD: 1 GLPI_CALDAV_IMPORT_STATE: 0 GLPI_DEMO_MODE: 0 GLPI_DOC_DIR: C:\xampp\htdocs\glpi10/files GLPI_CACHE_DIR: C:\xampp\htdocs\glpi10/files/_cache GLPI_CRON_DIR: C:\xampp\htdocs\glpi10/files/_cron GLPI_DUMP_DIR: C:\xampp\htdocs\glpi10/files/_dumps GLPI_GRAPH_DIR: C:\xampp\htdocs\glpi10/files/_graphs GLPI_LOCAL_I18N_DIR: C:\xampp\htdocs\glpi10/files/_locales GLPI_LOCK_DIR: C:\xampp\htdocs\glpi10/files/_lock GLPI_LOG_DIR: C:\xampp\htdocs\glpi10/files/_log GLPI_PICTURE_DIR: C:\xampp\htdocs\glpi10/files/_pictures GLPI_PLUGIN_DOC_DIR: C:\xampp\htdocs\glpi10/files/_plugins GLPI_RSS_DIR: C:\xampp\htdocs\glpi10/files/_rss GLPI_SESSION_DIR: C:\xampp\htdocs\glpi10/files/_sessions GLPI_TMP_DIR: C:\xampp\htdocs\glpi10/files/_tmp GLPI_UPLOAD_DIR: C:\xampp\htdocs\glpi10/files/_uploads GLPI_INVENTORY_DIR: C:\xampp\htdocs\glpi10/files/_inventories GLPI_NETWORK_REGISTRATION_API_URL: https://services.glpi-network.com/api/registration/ GLPI_MARKETPLACE_PLUGINS_API_URI: https://services.glpi-network.com/api/glpi-plugins/ GLPI_I18N_DIR: C:\xampp\htdocs\glpi10/locales GLPI_VERSION: 10.0.0-dev GLPI_SCHEMA_VERSION: 10.0.0-dev@5d5ed159d30f08ffa4fb9065d7c3f64855d5d4e1 GLPI_MARKETPLACE_PRERELEASES: 1 GLPI_MIN_PHP: 7.4.0 GLPI_YEAR: 2021

htmlawed/htmlawed version 1.2.6 in (C:\xampp\htdocs\glpi10\vendor\htmlawed\htmlawed) phpmailer/phpmailer version 6.5.3 in (C:\xampp\htdocs\glpi10\vendor\phpmailer\phpmailer\src) simplepie/simplepie version 1.5.6 in (C:\xampp\htdocs\glpi10\vendor\simplepie\simplepie\library) mpdf/mpdf in (C:\xampp\htdocs\glpi10\vendor\mpdf\mpdf\src) michelf/php-markdown in (C:\xampp\htdocs\glpi10\vendor\michelf\php-markdown\Michelf) true/punycode in (C:\xampp\htdocs\glpi10\vendor\true\punycode\src) iamcal/lib_autolink in (C:\xampp\htdocs\glpi10\vendor\iamcal\lib_autolink) sabre/dav in (C:\xampp\htdocs\glpi10\vendor\sabre\dav\lib\DAV) sabre/http in (C:\xampp\htdocs\glpi10\vendor\sabre\http\lib) sabre/uri in (C:\xampp\htdocs\glpi10\vendor\sabre\uri\lib) sabre/vobject in (C:\xampp\htdocs\glpi10\vendor\sabre\vobject\lib) laminas/laminas-i18n in (C:\xampp\htdocs\glpi10\vendor\laminas\laminas-i18n\src) laminas/laminas-servicemanager in (C:\xampp\htdocs\glpi10\vendor\laminas\laminas-servicemanager\src) monolog/monolog in (C:\xampp\htdocs\glpi10\vendor\monolog\monolog\src\Monolog) sebastian/diff in (C:\xampp\htdocs\glpi10\vendor\sebastian\diff\src) elvanto/litemoji in (C:\xampp\htdocs\glpi10\vendor\elvanto\litemoji\src) symfony/console in (C:\xampp\htdocs\glpi10\vendor\symfony\console) scssphp/scssphp in (C:\xampp\htdocs\glpi10\vendor\scssphp\scssphp\src) laminas/laminas-mail in (C:\xampp\htdocs\glpi10\vendor\laminas\laminas-mail\src\Protocol) laminas/laminas-mime in (C:\xampp\htdocs\glpi10\vendor\laminas\laminas-mime\src) rlanvin/php-rrule in (C:\xampp\htdocs\glpi10\vendor\rlanvin\php-rrule\src) blueimp/jquery-file-upload in (C:\xampp\htdocs\glpi10\vendor\blueimp\jquery-file-upload\server\php) ramsey/uuid in (C:\xampp\htdocs\glpi10\vendor\ramsey\uuid\src) psr/log in (C:\xampp\htdocs\glpi10\vendor\psr\log\Psr\Log) psr/simple-cache in (C:\xampp\htdocs\glpi10\vendor\psr\simple-cache\src) psr/cache in (C:\xampp\htdocs\glpi10\vendor\psr\cache\src) league/csv in (C:\xampp\htdocs\glpi10\vendor\league\csv\src) mexitek/phpcolors in (C:\xampp\htdocs\glpi10\vendor\mexitek\phpcolors\src\Mexitek\PHPColors) guzzlehttp/guzzle in (C:\xampp\htdocs\glpi10\vendor\guzzlehttp\guzzle\src) guzzlehttp/psr7 in (C:\xampp\htdocs\glpi10\vendor\guzzlehttp\psr7\src) glpi-project/inventory_format in (C:\xampp\htdocs\glpi10\vendor\glpi-project\inventory_format\lib\php) wapmorgan/unified-archive in (C:\xampp\htdocs\glpi10\vendor\wapmorgan\unified-archive\src) paragonie/sodium_compat in (C:\xampp\htdocs\glpi10\vendor\paragonie\sodium_compat\src) symfony/cache in (C:\xampp\htdocs\glpi10\vendor\symfony\cache) html2text/html2text in (C:\xampp\htdocs\glpi10\vendor\html2text\html2text\src) symfony/dom-crawler in (C:\xampp\htdocs\glpi10\vendor\symfony\dom-crawler) twig/twig in (C:\xampp\htdocs\glpi10\vendor\twig\twig\src) twig/string-extra in (C:\xampp\htdocs\glpi10\vendor\twig\string-extra) symfony/polyfill-ctype not found symfony/polyfill-iconv not found symfony/polyfill-mbstring not found symfony/polyfill-php80 in (C:\xampp\htdocs\glpi10\vendor\symfony\polyfill-php80)

Not active

Way of sending emails: SMTP (anonymous@srvexch)

 

reports              Name: Berichte                       Version: 1.13.1     State: Installed / not activated 	order                Name: Bestellverwaltung              Version: 2.6.0      State: Installed / not activated 	datainjection        Name: Data injection                 Version: 2.9.0      State: Installed / not activated 	formcreator          Name: Form Creator                   Version: 2.10.1     State: Installed / not activated 	accounts             Name: Konten                         Version: 2.6.0      State: Installed / not activated 	archires             Name: Network Architectures          Version: 2.7.0      State: Installed / not activated 	protocolsmanager     Name: Protocols manager              Version: 1.4        State: Installed / not activated

</body>
</html>

**Comment by cedric-anne** [comment]
Hi,

Could you check in your GLPI logs (`files/_log/php-errors.log`) to see if you have a more detailed trace of this error ?

Regards

**Comment by ukkt** [comment]
Hi,

this is every think:

[2021-12-16 06:19:12] glpiphplog.ALERT:   *** PHP Compile Error (64): Cannot declare class Glpi\Socket, because the name is already in use in C:\xampp\htdocs\glpi10\src\Socket.php at line 0  
[2021-12-16 06:19:40] glpiphplog.ALERT:   *** PHP Compile Error (64): Cannot declare class Glpi\Socket, because the name is already in use in C:\xampp\htdocs\glpi10\src\Socket.php at line 0  
[2021-12-16 06:19:44] glpiphplog.ALERT:   *** PHP Compile Error (64): Cannot declare class Glpi\Socket, because the name is already in use in C:\xampp\htdocs\glpi10\src\Socket.php at line 0  
[2021-12-16 06:19:46] glpiphplog.ALERT:   *** PHP Compile Error (64): Cannot declare class Glpi\Socket, because the name is already in use in C:\xampp\htdocs\glpi10\src\Socket.php at line 0  
[2021-12-16 06:19:50] glpiphplog.ALERT:   *** PHP Compile Error (64): Cannot declare class Glpi\Socket, because the name is already in use in C:\xampp\htdocs\glpi10\src\Socket.php at line 0  
[2021-12-16 06:19:52] glpiphplog.ALERT:   *** PHP Compile Error (64): Cannot declare class Glpi\Socket, because the name is already in use in C:\xampp\htdocs\glpi10\src\Socket.php at line 0  
[2021-12-16 06:19:55] glpiphplog.ALERT:   *** PHP Compile Error (64): Cannot declare class Glpi\Socket, because the name is already in use in C:\xampp\htdocs\glpi10\src\Socket.php at line 0  
[2021-12-16 06:19:56] glpiphplog.ALERT:   *** PHP Compile Error (64): Cannot declare class Glpi\Socket, because the name is already in use in C:\xampp\htdocs\glpi10\src\Socket.php at line 0  
[2021-12-16 06:19:57] glpiphplog.ALERT:   *** PHP Compile Error (64): Cannot declare class Glpi\Socket, because the name is already in use in C:\xampp\htdocs\glpi10\src\Socket.php at line 0  
[2021-12-16 06:19:59] glpiphplog.ALERT:   *** PHP Compile Error (64): Cannot declare class Glpi\Socket, because the name is already in use in C:\xampp\htdocs\glpi10\src\Socket.php at line 0  
[2021-12-16 06:20:03] glpiphplog.ALERT:   *** PHP Compile Error (64): Cannot declare class Glpi\Socket, because the name is already in use in C:\xampp\htdocs\glpi10\src\Socket.php at line 0  
[2021-12-16 06:20:05] glpiphplog.ALERT:   *** PHP Compile Error (64): Cannot declare class Glpi\Socket, because the name is already in use in C:\xampp\htdocs\glpi10\src\Socket.php at line 0  

When I start the debug mode, I get the error on all assets. The globle view is ok.

**Comment by ukkt** [comment]
Also the plugins show the error.

**Comment by Deft-54** [comment]
Hello,
I've got the same error in 'Computers', 'network devices', 'phones', 'crontasks', and some other categories.
Thanks in advance

**Comment by cedric-anne** [comment]
If anyone is able to produce a detailed stacktrace, it would help. Also, can you test with no active plugins and give your system informations ?

**Comment by Deft-54** [comment]
> If anyone is able to produce a detailed stacktrace, it would help. Also, can you test with no active plugins and give your system informations ?

Sorry I don't know how to produce a 'detailed stacktrace'... Can you tell me please what you what or what to do ? Is it the content of a some files in ..files/_log/ ? I'll do what you want if it can help, just guide me a bit please

**Comment by ukkt** [comment]
> If anyone is able to produce a detailed stacktrace, it would help. Also, can you test with no active plugins and give your system informations ?

I have got full trace and debug mode on and I only get this lines.

**Comment by Deft-54** [comment]
> > If anyone is able to produce a detailed stacktrace, it would help. Also, can you test with no active plugins and give your system informations ?
> 
> I have got full trace and debug mode on and I only get this lines.

me too I just checked a few minutes ago

**Comment by cedric-anne** [comment]
see #10168 for stacktrace

**Comment by Deft-54** [comment]
Thank you very much !!
No more issue in computer / phone / network devices...
Best regards ;-)

**Comment by trasher** [comment]
Fixed in #10169

**Comment by ukkt** [comment]
Thank you!!
